pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1430,36 +1445,31 @@ func (ctrl *PersistentVolumeController) provisionClaim(claim *v1.PersistentVolum
 		// retain the original behavior of returning nil from provisionClaim call
 		return nil
 	}
-	// for CIS provisioning, latency metric will be reported after final binding happened
-	if plugin == nil || plugin.IsMigratedToCSI() {
-		// TEST case
-		ctrl.scheduleOperation(opName, func() error {
-			// create an opTimestamps without assigning provisionerName to get the current timestamp
-			opTimestamps := newOperationTimestamp("")
-			provisionerName, err := ctrl.provisionClaimOperationCSI(claim, plugin, storageClass)
-			opTimestamps.provisionerName = provisionerName
-			// adding an entry into operationTimestamps for metrics reporting if there exists no
-			// such record for current claim regardless whether the operation returned error or not
-			// add an entry to mark the provisioning request start time
-			claimKey := claimToClaimKey(claim)
-			provisionStartTsKey := ctrl.createOperationTimestampCacheKey("provision", claimKey)
-			if _, exists := ctrl.operationTimestamps.Get(provisionStartTsKey); !exists {
-				ctrl.operationTimestamps.Add(provisionStartTsKey, opTimestamps)
-			}
-			return err
-		})
-	} else /* plugin != nil && !plugin.IsMigratedToCSI() */ {
-		// record provision start time for operation metric reporting
-		startTime := time.Now()
-		ctrl.scheduleOperation(opName, func() error {
-			pluginName, err := ctrl.provisionClaimOperation(claim, plugin, storageClass)
-			// TODO: timeTaken here ONLY counts the first-try time and may not necessarily
-			//       represent the real end to end latency of provisioning
-			timeTaken := time.Since(startTime).Seconds()
-			metrics.RecordVolumeOperationMetric(pluginName, "provision", timeTaken, err)
-			return err
-		})
-	}
+	ctrl.scheduleOperation(opName, func() error {
+		// create an opTimestamps without assigning provisionerName/pluginName to get the current timestamp
+		opTimestamps := newOperationTimestamp("")
+		var provisionerName string
+		var err error
+		if plugin == nil || plugin.IsMigratedToCSI() {
```

Comment:
@ddebroy can you review this to see if this affects your changes

Reply:
accommodated @ddebroy comments. resolving